File: Interview Preparation Kit - Python/01. Warm-up Challenges/001. Sock Merchant.py
# Problem: https://www.hackerrank.com/challenges/sock-merchant/problem
# Score: 10


def sock_merchant(ar):
    list_of_socks = [0 for i in range(101)]
    for element in ar:
        list_of_socks[element] = list_of_socks[element] + 1
    ans = 0
    for sock in list_of_socks:
        ans += sock // 2
    return ans


n = int(input().strip())
ar = list(map(int, input().strip().split(' ')))
result = sock_merchant(ar)
print(result)

File: Interview Preparation Kit - CPP/01. Warm-up Challenges/001. Sock Merchant.cpp
// Problem: https://www.hackerrank.com/challenges/sock-merchant/problem
// Score: 10


#include <iostream>


int main() {
    int n, element;
    int arr[101] = {0};
    int ans = 0;

    std::cin >> n;

    for (int i = 0; i < n; i++) {
        std::cin >> element;
        arr[element]++;

        ans += arr[element] / 2;
        arr[element] = arr[element] % 2;
    }

    std::cout << ans << std::endl;
    return 0;
}
